### PR TITLE
HOCS-2310 Running custom reports takes too long

### PR DESCRIPTION
**Pull request by agraham19**
* Add cron job to refresh dcu aggregated cases view

### CODE REVIEW

File: kd/refreshDcuAggregatedCasesView.yaml [hunk 1]
```diff
@@ -0,0 +1,31 @@
+apiVersion: batch/v1beta1
+kind: CronJob
+metadata:
+  name: hocs-refresh-dcu-aggregated-cases-view-job
+spec:
+  schedule: "0 6 * * 1-5"
```
Comment (zuzak) [inline kd/refreshDcuAggregatedCasesView.yaml:6]: If the cron is always in UTC (likely), this will fire at 7am Europe/London half the year. If that's intended, it's fine -- but I think people are using the system at 7am. Maybe we should fire the job late the evening before (`0 21`?) The refresh members job apparently fires at 22:00Z.

File: kd/refreshDcuAggregatedCasesView.yaml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+              image: quay.io/ukhomeofficedigital/openjdk11:v11.0.5_10
+              env:
+                - name: HOCS_BASICAUTH
+                  valueFrom:
+                    secretKeyRef:
+                      name: ui-casework-creds
+                      key: plaintext
+              command: ["/bin/sh", "-c"]
+              args:
+                - http_status=$( curl -vk -u ${HOCS_BASICAUTH} -w '%{http_code}' https://hocs-audit.{{.KUBE_NAMESPACE}}.svc.cluster.local/export/custom/DCU_AGGREGATED_CASES/refresh ); if [[ $http_status -eq 200 ]]; then exit 0; else exit 1; fi
```
Comment (zuzak) [inline kd/refreshDcuAggregatedCasesView.yaml:30]: You can use [YAML multiline strings](https://yaml-multiline.info/) to split this over multiple lines, instead of one big one.

File: kd/refreshDcuAggregatedCasesView.yaml [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+          restartPolicy: OnFailure
```
Comment (zuzak) [inline kd/refreshDcuAggregatedCasesView.yaml:31]: Finish this file with a new-line if you can.

File: kd/refreshDcuAggregatedCasesView.yaml [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+          labels:
+            name: hocs-refresh-dcu-aggregated-cases-view-job
+            role: hocs-backend
+        spec:
+          containers:
+            - name: hocs-refresh-dcu-aggregated-cases-view-job
+              securityContext:
+                runAsNonRoot: true
+                runAsUser: 1000
+              image: quay.io/ukhomeofficedigital/openjdk11:v11.0.5_10
```
Comment (zuzak) [inline kd/refreshDcuAggregatedCasesView.yaml:21]: This is quite an over-weight container -- it spins up the entire JVM and then doesn't use it. If you can, use a leaner image with just curl installed. (If you can't, don't worry, but add a note to HOCS-2141.)

File: kd/refreshDcuAggregatedCasesView.yaml [hunk 5]
```diff
@@ -0,0 +1,31 @@
+apiVersion: batch/v1beta1
+kind: CronJob
+metadata:
+  name: hocs-refresh-dcu-aggregated-cases-view-job
```
Comment (zuzak) [inline kd/refreshDcuAggregatedCasesView.yaml:4]: I would consider shortening these names, just because they'll make `kubectl` output very long.
The existing cron is called `hocs-refresh-members-job`; something about that length would be nice (`hocs-refresh-dcu-caseview` maybe?)

**Review by zuzak — APPROVED** [review]
I would rephrase your commit message so that it's obvious from `git log` what you're doing.
For example:

```gitcommit
HOCS-2310 Add cron to refresh DCU agg case view

This commit adds a cronjob to hit the workflow endpoint added
to hocs-workflow in [commit sha] every morning at 6am.

We picked once a day at 6am because X. 
```

File: kd/refreshDcuAggregatedCasesView.yaml (hunk 1 above)
Comment (agraham19) [reply kd/refreshDcuAggregatedCasesView.yaml]: I have changed this to 22:00

File: kd/refreshDcuAggregatedCasesView.yaml (hunk 3 above)
Comment (agraham19) [reply kd/refreshDcuAggregatedCasesView.yaml]: I have addressed this

File: kd/refreshDcuAggregatedCasesView.yaml (hunk 5 above)
Comment (agraham19) [reply kd/refreshDcuAggregatedCasesView.yaml]: I have made this change

**Comment by zuzak** [comment]
Switching to draft to prevent accidental merge.
